ergo-wallet/src/main/scala/org/ergoplatform/wallet/transactions/TransactionBuilder.scala

```diff
@@ -19,10 +18,144 @@ import scorex.crypto.hash.Digest32
 import org.ergoplatform.wallet.{AssetUtils, TokensMap}
 import org.ergoplatform.wallet.boxes.BoxSelector
 import org.ergoplatform.wallet.boxes.DefaultBoxSelector
-
+import scorex.crypto.authds.ADKey
+import scorex.util.encode.Base16
+import scala.collection.JavaConverters._
 
 object TransactionBuilder {
 
+  /**
+    * @param recipientAddress - payment recipient address
+    * @param transferAmt - amount of ERGs to transfer
+    */
+  case class Payment(
+    recipientAddress: ErgoAddress,
+    transferAmt: Long
+  )
+
+  /**
+    * Assembles unsigned payment transaction with multiple outputs
+    *
+    * @param inputIds - identifiers of inputs to be used in transaction
+    * @param feeAmt - fee amount
+    * @param currentHeight - current blockchain height
+    * @param payments - list of addresses and corresponding amounts to make outputs from
+    * @param changeAddress - change recipient address
+    * @param changeAmt - amount to return back to `changeAddress`
+    * @return unsigned transaction
+    */
+  def multiPaymentTransaction(inputIds: Array[String],
+                              feeAmt: Long,
+                              payments: java.util.List[Payment],
+                              changeAddress: ErgoAddress,
+                              changeAmt: Long,
+                              currentHeight: Int): UnsignedErgoLikeTransaction = {
+    val outputs =
+      payments.asScala.flatMap { case Payment(recipientAddress, transferAmt) =>
+        val payTo = new ErgoBoxCandidate(
+          transferAmt,
+          recipientAddress.script,
+          currentHeight,
+          Seq.empty[(ErgoBox.TokenId, Long)].toColl,
+          Map.empty
+        )
+        val fee = new ErgoBoxCandidate(
```

Comment:
You still create a fee box and change box on every payment output!

Reply:
fixed https://github.com/ergoplatform/ergo/pull/1706/commits/46f0e45eacf10074f831368f1ab50ee33c34ad9e